### PR TITLE
Take new dev workflow changes from CoreFx

### PR DESCRIPTION
**Pull request by roncain**
This PR moves WCF to match CoreFx in the way build tools are restored, which has changed substantially.  It now installs the CLI and uses 'dotnet' rather than dnu to restore packages.

Running build.sh on Ubuntu succeeded for me without timeouts.

I would like to defer merging this until @StephenBonikowsky has added package projects unless that work requires the changes in this PR be made first.

/cc @StephenBonikowsky @mconnew @iamjasonp

### CODE REVIEW

**Comment by roncain** [comment]
Reviewers -- it may become necessary for this PR to originate in TFS because there are compensating changes required in TFS itself.  So please comment but please do not merge without a conversation.

File: dir.targets [hunk 1]
```diff
@@ -1,120 +1,20 @@
...
-           AlwaysCreate="true"
-           ForceTouch="true" />
-
-    <Error Condition="'$(ErrorIfBuildToolsRestoredFromIndividualProject)'=='true'"
-           Text="The build tools package was just restored and so we cannot continue the build of an individual project because targets from the build tools package were not able to be imported. Please retry the build the individual project again." />
+  <Target Name="CheckForBuildTools">
+    <Error Condition="!Exists('$(ToolsDir)') and '$(OverrideToolsDir)'=='true'"
+           Text="The tools directory [$(ToolsDir)] does not exist. Please run sync in your enlistment to ensure the tools are installed before attempting to build an individual project." />
+    <Error Condition="!Exists('$(ToolsDir)') and '$(OverrideToolsDir)'!='true'"
+           Text="The tools directory [$(ToolsDir)] does not exist. Please run init-tools.cmd in your enlistment to ensure the tools are installed before attempting to build an individual project." />
```
Comment (iamjasonp) [inline dir.targets]: These two errors are exactly the same condition but different messages. Is this intentional to draw attention to the need to both `sync` and `init-tools` ?

Comment (roncain) [inline dir.targets]: I should have mentioned these edits are direct copies from CoreFx.  I'll raise the question to the CoreFx authors but would like to stay in sync.

**Comment by roncain** [comment]
@StephenBonikowsky @mconnew and @iamjasonp -- could I ask you to review this?  I have buddy built it in TFS and Linux.  I'll need to rebase with Stephen's PR merge, and I will do the TFS build again, but I would like to have the option to merge asap so we don't have to go to shiproom for permission (starting tomorrow morning).  Virtually all changes here are just lifted verbatim from CoreFx.

File: dir.props [hunk 2]
```diff
@@ -56,9 +56,22 @@
     <RuntimeIdGraphDefinitionFile>$(ProjectDir)pkg/Microsoft.NETCore.Platforms/runtime.json</RuntimeIdGraphDefinitionFile>
```
Comment (StephenBonikowsky) [inline dir.props]: Rebasing after my PR is on will fix this line.

**Comment by StephenBonikowsky** [comment]
LGTM
I can do a buddy build as well as soon as #694 is merged.

File: dir.targets (hunk 1 above)
Comment (mconnew) [inline dir.targets]: They aren't quite identical conditions. In the second clause one is == and other  !=

Comment (iamjasonp) [inline dir.targets]: Ah, thanks @mconnew

**Comment by iamjasonp** [comment]
LGTM, buddy building this.

**Comment by iamjasonp** [comment]
Buddy built inner and outerloop - all looks good

**Comment by roncain** [comment]
@dotnet-bot test outerloop please

**Pull request merged**